src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkCapacity();
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexRangeInAdd(index);
+        checkCapacity();
+        checkElementByIndex(index);
+        elements[index] = value;
+        size++;
```

Comment:
The method 'add(T value, int index)' has a redundant call to 'checkElementByIndex(index)'. Instead, you should use 'System.arraycopy()' to shift the elements after the insertion point. Also, you are not actually inserting the new element into the array; you are overwriting the element at the index.